server/setup_flow.go

```diff
@@ -633,9 +633,20 @@ func (p *Plugin) submitCreateCloudOAuthInstance(f *flow.Flow, submission map[str
 		return "", nil, nil, errors.New("no Jira OAuth Client Secret is present in the request")
 	}
 
-	jiraURL, instance, err := p.installCloudOAuthInstance(jiraURL, clientID, clientSecret)
+	existingInstance, err := p.instanceStore.LoadInstance(types.ID(jiraURL))
 	if err != nil {
-		return "", nil, nil, err
+		return "", nil, nil, errors.Wrap(err, "failed to load existing cloud-oauth instance")
+	}
+
+	instance, ok := existingInstance.(*cloudOAuthInstance)
+	if !ok {
+		return "", nil, nil, errors.Errorf("existing instance is not a cloud-oauth instance. ID: %s", jiraURL)
+	}
+
+	instance.JiraClientID = clientID
+	instance.JiraClientSecret = clientSecret
+	if err = p.instanceStore.StoreInstance(instance); err != nil {
```

Comment:
Why are we storing the clientID and clientSecret of the new oauth instance to the existing one?

Reply:
There's not really a "new" instance here. There will always be an existing one in the kv store at this point of the setup. There are multiple stages to the setup flow and this is the second one